shared_model/validators/field_validator.hpp

```diff
@@ -160,33 +187,42 @@ namespace shared_model {
         iroha::ts64_t now = iroha::time::now();
         // TODO 06/08/17 Muratov: make future gap for passing timestamp, like
         // with old timestamps IR-511 #goodfirstissue
+
+        auto time_message =
+            (boost::format("%llu, now: %llu") % timestamp % now).str();
         if (now < timestamp) {
-          reason.second.push_back(boost::str(
-              boost::format(
-                  "timestamp broken: send from future (%llu, now %llu)")
-              % timestamp % now));
+          auto message = generateErrorMessage(
+              "timestamp broken: sent from future", time_message);
+          reason.second.push_back(std::move(message));
         }
 
-        if (now - timestamp > MAX_DELAY) {
-          reason.second.push_back(boost::str(
-              boost::format("timestamp broken: too old (%llu, now %llu)")
-              % timestamp % now));
+        if (now - timestamp > max_delay) {
+          auto message =
+              generateErrorMessage("timestamp broken: too old", time_message);
+          reason.second.push_back(std::move(message));
         }
       }
 
       void validateCounter(ReasonsGroupType &reason,
                            const interface::types::CounterType &counter) const {
         if (counter == 0) {
-          reason.second.push_back(
-              boost::str(boost::format("Counter should be > 0")));
+          reason.second.push_back("Counter should be > 0");
         }
       }
 
      private:
       std::regex account_id_, asset_id_, ip_address_, name_, detail_key_;
       // max-delay between tx creation and validation
-      static constexpr auto MAX_DELAY =
+      static constexpr auto max_delay =
           std::chrono::hours(24) / std::chrono::milliseconds(1);
+      // size of key
+      static constexpr auto key_size = 32;
+
+      std::string generateErrorMessage(const std::string &error,
```

Comment:
Now I am thinking that this method is not needed at all, thus making a developer who writes a validator responsible for the message.

Reply:
I had similar thoughts, it just makes everything more complicated, so I'll remove it.